cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_common.rs

```diff
@@ -0,0 +1,524 @@
+			AssetHubWestendUniversalLocation,
+			[u8; 32],
+		>::convert_location(&Location::new(
+				2,
+				[xcm::v5::Junction::GlobalConsensus(EthereumNetwork::get())],
+			))
+			.unwrap()
+			.into()
+	});
+
+	ethereum_sovereign
+}
+
+pub fn weth_location() -> Location {
+	erc20_token_location(WETH.into())
+}
+
+pub fn eth_location() -> Location {
+	Location::new(2, [GlobalConsensus(Ethereum { chain_id: CHAIN_ID })])
+}
+
+pub fn ethereum() -> Location {
+	eth_location()
+}
+
+pub fn erc20_token_location(token_id: H160) -> Location {
+	Location::new(
+		2,
+		[
+			GlobalConsensus(EthereumNetwork::get().into()),
+			AccountKey20 { network: None, key: token_id.into() },
+		],
+	)
+}
+
+// ROC and wROC
+pub(crate) fn roc_at_ah_rococo() -> Location {
+	Parent.into()
+}
+pub(crate) fn bridged_roc_at_ah_westend() -> Location {
+	Location::new(2, [GlobalConsensus(ByGenesis(ROCOCO_GENESIS_HASH))])
+}
+
+pub(crate) fn create_foreign_on_ah_westend(id: xcm::opaque::v5::Location, sufficient: bool) {
+	let owner = AssetHubWestend::account_id_of(ALICE);
+	AssetHubWestend::force_create_foreign_asset(id, owner, sufficient, ASSET_MIN_BALANCE, vec![]);
+}
+
+// set up pool
+pub(crate) fn set_up_pool_with_wnd_on_ah_westend(
+	asset: Location,
+	is_foreign: bool,
+	initial_fund: u128,
+	initial_liquidity: u128,
+) {
+	let wnd: Location = Parent.into();
+	AssetHubWestend::fund_accounts(vec![(AssetHubWestendSender::get(), initial_fund)]);
+	AssetHubWestend::execute_with(|| {
+		type RuntimeEvent = <AssetHubWestend as Chain>::RuntimeEvent;
+		let owner = AssetHubWestendSender::get();
+		let signed_owner = <AssetHubWestend as Chain>::RuntimeOrigin::signed(owner.clone());
+
+		if is_foreign {
+			assert_ok!(<AssetHubWestend as AssetHubWestendPallet>::ForeignAssets::mint(
+				signed_owner.clone(),
+				asset.clone().into(),
+				owner.clone().into(),
+				initial_fund,
+			));
+		} else {
+			let asset_id = match asset.interior.last() {
+				Some(GeneralIndex(id)) => *id as u32,
+				_ => unreachable!(),
+			};
+			assert_ok!(<AssetHubWestend as AssetHubWestendPallet>::Assets::mint(
+				signed_owner.clone(),
+				asset_id.into(),
+				owner.clone().into(),
+				initial_fund,
+			));
+		}
+		assert_ok!(<AssetHubWestend as AssetHubWestendPallet>::AssetConversion::create_pool(
+			signed_owner.clone(),
+			Box::new(wnd.clone()),
+			Box::new(asset.clone()),
+		));
+		assert_expected_events!(
+			AssetHubWestend,
+			vec![
+				RuntimeEvent::AssetConversion(pallet_asset_conversion::Event::PoolCreated { .. }) => {},
+			]
+		);
+		assert_ok!(<AssetHubWestend as AssetHubWestendPallet>::AssetConversion::add_liquidity(
+			signed_owner.clone(),
+			Box::new(wnd),
+			Box::new(asset),
+			initial_liquidity,
+			initial_liquidity,
+			1,
+			1,
+			owner.into()
+		));
+		assert_expected_events!(
+			AssetHubWestend,
+			vec![
+				RuntimeEvent::AssetConversion(pallet_asset_conversion::Event::LiquidityAdded {..}) => {},
+			]
+		);
+	});
+}
```

Comment:
Can't you use [this](https://github.com/acatangiu/polkadot-sdk/blob/64dc0effc3fe7565667c93d9a797bb31c5c6bfc9/cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/mod.rs#L116) instead?